syntax = 'proto3';

package fixbug;

option cc_generic_services = true;

message ResultCode
{
    int32 errcode = 1;
    bytes errmsg = 2;
}

message LoginRequest   //定义函数参数类型
{
    bytes name = 1;
    bytes pwd = 2;
}

message LoginResponse  //定义函数返回值类型
{
    ResultCode result = 1;
    bool success = 2;
}

message RegisterRequest
{
    uint32 id = 1;
    bytes name = 2;
    bytes pwd = 3;
}

message RegisterResponse  //定义函数返回值类型
{
    ResultCode result = 1;
    bool success = 2;
}

service UserServiceRpc  
{
    rpc Login(LoginRequest) returns(LoginResponse);  //rpc函数的描述
    rpc Register(RegisterRequest) returns(RegisterResponse);
}